src/__tests__/provider.test.tsx

```diff
@@ -0,0 +1,40 @@
+import { render } from "@testing-library/react";
+import { createBrowserHistory } from "history";
+import React, { ReactElement } from "react";
+import { act } from "react-dom/test-utils";
+
+import { Auth0Provider } from "../provider";
+
+describe("Auth0Provider", () => {
+  describe("renderLoading view", () => {
+    it("renders null when no loading view is supplied", async () => {
+      await act(async () => {
+        const { container } = await render(
+          <Auth0Provider client_id={""} domain={""} history={createBrowserHistory()}>
+            <div>contents</div>
+          </Auth0Provider>,
+        );
+
+        // console.log(x.container.childElementCount);
+        expect(container.childElementCount).toEqual(0);
+      });
+    });
+
+    it("renders the loading view when supplied", async () => {
+      await act(async () => {
```

Comment:
what is the act used for here? is it needed?

Reply:
I believe it's due to the async work happening on the render.  (I could be wrong!) See previous comment response from me... this is basically the framework telling me I need it -- I don't have a complete understanding of the implication here myself as of yet.